python/ffsim/tenpy/hamiltonians/molecular_hamiltonian.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import numpy as np
+from tenpy.models.model import CouplingMPOModel
+from tenpy.networks.site import SpinHalfFermionSite
+
+from ffsim.hamiltonians.molecular_hamiltonian import MolecularHamiltonian
+from ffsim.tenpy.hamiltonians.lattices import MolecularChain
+
+# ignore lowercase variable checks to maintain TeNPy naming conventions
+# ruff: noqa: N806
+
+
+class MolecularHamiltonianMPOModel(CouplingMPOModel):
+    """Molecular Hamiltonian."""
+
+    def __init__(self, params):
+        if hasattr(self, "flag"):  # only call __init__ once
+            return
+        self.flag = True
```

Comment:
I was actually meaning to ask you about this, because I am not sure. The `__init__` function is needed for consistency with TeNPy code conventions and it can only be called once (because it adds terms to the Hamiltonian). In fact, if you try to call it twice, TeNPy will error because then you will duplicate all of the terms in the Hamiltonian. Now that I have moved `from_molecular_hamiltonian` to a static method within the class, there is a line within this method where I need to create a `MolecularHamiltonianMPOModel` instance. Therefore, there is now a chance that `__init__` will be called twice. To resolve this, I have added a simple flag to `__init__`, which makes sure that it is only executed once (on the first time through). Please let me know if you have a more elegant solution :pray:

Reply:
Sorry, I still don't understand. Can you give me a code example that would cause `__init__` to be called twice?

Reply:
You're right, I cannot reproduce this anymore. This issue must have resolved itself during the restructuring somehow. I have removed these lines now.